### ISSUE TITLE
ios-webkit-proxy missing under IOSServerFlag

### ISSUE THREAD

**Issue by saikrishna321** [appium/java-client#343]
@TikhomirovSergey it's good to have --webkit-debug-proxy-port under the IOSServerFlag as we are running our test in parallel and it's a hybrid app, i'm in need to start the appium server with this argument.

Appium 1.5v has this argument, please look at the link below 

https://github.com/appium/appium/blob/master/lib/parser.js#L338

**Comment by TikhomirovSergey** [comment]
Yes. I've noticed it.

**Comment by saikrishna321** [comment]
:+1:

**Comment by VinitSandesara** [comment]
Hi https://github.com/TikhomirovSergey, Any ETA for this enhancement will be released ? we recently have added new hybrid feature in apps and we have very close deadline to finished implementing that feature but due to this we are stuck now.

Thanks
VInit

**Comment by TikhomirovSergey** [comment]
The build 3.4.1 is coming soon.

@VinitSandesara 
You can do that

```java
ServerArgument argument = new ServerArgument() {
    @Override
    public String getArgument() {
        return "--webkit-debug-proxy-port";
    }
};
service = new AppiumServiceBuilder().withArgument(argument, "27753").build();
```

untill the build is published

**Comment by VinitSandesara** [comment]
Thanks you so much TikhomirovSergey for making this available to us. We will add this solution to our framework.

Thanks
Vinit

**Comment by VinitSandesara** [comment]
Hi TikhomirovSergey, i have added this change in our framework and noticed that 1/10 times it is getting stuck on "driver.context(contextName);" while switching to Webview from Native, what could be the cause of this issue.

![screen shot 2016-03-24 at 12 46 29 pm](https://cloud.githubusercontent.com/assets/12589206/14026004/bdb9c04e-f1be-11e5-9f36-2415f2068140.png)

Thanks
Vinit

**Comment by TikhomirovSergey** [comment]
@VinitSandesara 
Please it is another issue. It is something wrong on the server side probably. 
Please report the issue here: https://github.com/appium/appium/issues. Also please report an issue here with the attached link to the server-side bug report.

**Comment by VinitSandesara** [comment]
@TikhomirovSergey , I have raised an issue against Appium server but noticed another issue if i assign port 27753 with ios_webkit_debug_proxy, after switching from Native to Webview next line of web action code works fine (if it is click, it click on the element) but if i assign any other port with ios_webkit_debug_proxy, after switching from Native to Webview next line of web action code doesn't work (if it is click, it doesn't click on the element and execution take forever until manually killed an execution).

Thanks
Vinit

**Comment by VinitSandesara** [comment]
@TikhomirovSergey, Any update on above comment ? with any port other then 27753 assigned with ios_webkit_debug_proxy doesn't perform any web view action once switched to web view from native.

Thanks
Vinit

**Comment by saikrishna321** [comment]
@VinitSandesara i see WEBVIEW_3 and WEBVIEW_4 in your logs .. are you switching to the right contextViews?

**Comment by VinitSandesara** [comment]
@saikrishna321 , I am switching to webview from Native context as soon as it finds from list. it is not required which webview to switch for the web element to click, send keys etc. I have already tested this same scenario in sample test where this is working fine but there i am manually connecting to iOs webkit proxy server using command on terminal. 

Thanks
Vinit
